### PR TITLE
Refs #29324 -- Fixed SECRET_KEY validation when using settings.configure().

### PR DESCRIPTION
**Pull request by timgraham**
Regression in b3cffde5559c4fa97625512d7ec41a674be26076.
https://code.djangoproject.com/ticket/29324#comment:5

### CODE REVIEW

File: django/conf/__init__.py [hunk 1]
```diff
@@ -56,6 +56,8 @@ def __getattr__(self, name):
         if self._wrapped is empty:
             self._setup(name)
         val = getattr(self._wrapped, name)
+        if name == 'SECRET_KEY' and not val:
+            raise ImproperlyConfigured('The SECRET_KEY setting must not be empty.')
```
Comment (claudep) [inline django/conf/__init__.py]: I'm not sure to like the additional test for every setting access, for a corner-case use case. I think we should consider reverting the initial patch instead.

Comment (timgraham) [reply django/conf/__init__.py]: I think the checks could be moved to the `configure()` method. Although the original commit still patches a `__getattr__()` method and may have a performance impact. I'm fine with reverting if an alternative isn't proposed.

**Comment by timgraham** [comment]
I'll revert the original patch for now.

**Comment by jdufresne** [comment]
Sorry, I've been away and was unable to respond earlier.

I'm fine with reverting the patch. I'll see if I can com up with a better solution.

**Pull request closed without merging**